Pod/Core/XCTestCase+Gherkin.swift

```diff
@@ -33,20 +35,98 @@ class GherkinState {
-    let stepChecker = GherkinStepsChecker()
+    fileprivate var missingStepsImplementations = [String]()
+    
+    func gherkinStepsAndMatchesMatchingExpression(_ expression: String) -> [(step:Step, match:NSTextCheckingResult)] {
+        let range = NSMakeRange(0, expression.characters.count)
+        let matches = self.steps.map { (step: Step) -> (step:Step, match:NSTextCheckingResult)? in
+            if let match = step.regex.firstMatch(in: expression, options: [], range: range) {
+                return (step:step, match:match)
+            } else {
+                return nil
+            }
+        }.flatMap { $0! }
+        return matches
+    }
+    
+    func gherkinStepsMatchingExpression(_ expression: String) -> [Step] {
+        return self.gherkinStepsAndMatchesMatchingExpression(expression).map { $0.step }
+    }
+    
+    // TODO: This needs to be refactored since function has a side effect (appends to missingStepsImplementations)
+    func matchingGherkinStepExpressionFound(_ expression: String) -> Bool {
+        let matches = self.gherkinStepsMatchingExpression(expression)
+        switch matches.count {
+            
+        case 0:
+            print("Step definition not found for '\(ColorLog.red(expression))'")
+            let stepImplementation = "step(\"\(expression)"+"\") {XCTAssertTrue(true)}"
+            self.missingStepsImplementations.append(stepImplementation)
+            
+        case 1:
+            //no issues, so proceed
+            return true
+            
+        default:
+            matches.forEach { NSLog("Matching step : \(String(reflecting: $0))") }
+            print("Multiple step definitions found for : '\(ColorLog.red(expression))'")
+        }
+        return false
+    }
+    
+    func shouldPrintTemplateCodeForAllMissingSteps() -> Bool {
+        return self.missingStepsImplementations.count > 0
+    }
+    
+    func resetMissingSteps() {
+        self.missingStepsImplementations = []
+    }
+    
+    func printTemplatedCodeForAllMissingSteps() {
+        print(ColorLog.red("Copy paste these steps in a StepDefiner subclass:"))
+        print("-------------")
+        self.missingStepsImplementations.forEach({
+            print($0)
+        })
+        print("-------------")
+    }
+    
+    func printStepDefinitions() {
+        self.loadAllStepsIfNeeded()
+        print("-------------")
+        print(ColorLog.darkGreen("Defined steps"))
+        print("-------------")
+        print(self.steps.map { String(reflecting: $0) }.sorted { $0.lowercased() < $1.lowercased() }.joined(separator: "\n"))
+        print("-------------")
+    }
+    
+    func loadAllStepsIfNeeded() {
+        guard self.steps.count == 0 else { return }
+        
+        // Create an instance of each step definer and call it's defineSteps method
+        allSubclassesOf(StepDefiner).forEach { subclass in
```

Comment:
It is not even my code to start with, just moved it... but a good point anyway. Fixed! :)